src/epochs.lua

```diff
@@ -586,7 +586,9 @@ function epochs.distributeRewardsForEpoch(currentTimestamp)
 					local actualDelegateReward = math.floor(eligibleDelegateReward * percentOfEligibleEarned)
 					-- distribute the rewards to the delegate if greater than 0
 					if actualDelegateReward > 0 then
-						balances.transfer(delegateAddress, ao.id, actualDelegateReward)
+						-- increase the stake and decrease the protocol balance
+						gar.increaseExistingDelegateStake(gatewayAddress, delegateAddress, actualDelegateReward)
```

Comment:
Every call to this is going to create a copy of the gateway object. Might want to factor out fetching the gateway object outside the loop and then pass the gw into the function to save on wasteful compute.

Reply:
Might be worth a search around the contract for patterns like that too... i.e. wasteful copying in loops.

Reply:
refer to the changes here - https://github.com/ar-io/ar-io-network-process/pull/146/commits/dcaa064d7d3d28cbac25e0909b5c8e0e5e6dc739

reduces duplicate copies within this handler - the call signature comments help with static analysis

Reply:
Looks great, thanks!